### PR TITLE
Add contextualIdentities.move

### PR DESCRIPTION
**Pull request by rebloor**
### Description

Documentation and release note in support of [Bug 1333395](https://bugzilla.mozilla.org/show_bug.cgi?id=1333395) Consider support for containers sort order.

### CODE REVIEW

File: files/en-us/mozilla/add-ons/webextensions/api/contextualidentities/move/index.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+- `position`
+  - : `integer`. The position to move `cookieStoreIds` to in the list of contextual identities. `-1` indicates that the items are moved to the end of the list.
+
+### Return value
+
+A [`Promise`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) that is fulfilled with an array of cookie store IDs representing the re-ordered list. If the request is for an invalid move or the contextual identities feature is not enabled, the promise is rejected.
+
+## Examples
+
+Starting with this list of contextual identities `[id0, id1, id2, id3]`:
```
Comment (Rob--W) [inline files/en-us/mozilla/add-ons/webextensions/api/contextualidentities/move/index.md:36]: I guess that these were based on the testcase. In there, id0, id1, etc are variable names referring to the actual IDs. The examples below are unclear and not working.

I would simplify this section by showing how one can query identities, then choose the identity to move. For example, the scenario of moving the last identity to the start, and then back to the end. And another one where multiple identities are moved.

```js
let identities = await browser.contextualIdentities.query({});
let firstId = identities[0].cookieStoreId;

// Moves first identity to the end.
await browser.contextualIdentities.move(firstId, -1);

// Move identity to the start again.
await browser.contextualIdentities.move(firstId, 0);
```

Alternative way of moving the first identity to the end: by moving all other identities to the start we get the same effect:

```js
let identities = await browser.contextualIdentities.query({});
let ids = identities.map(identity => identity.cookieStoreId);
// Create array without the first item:
let otherIds = ids.slice(1);

// Moves other identities to the start,
// effectively putting the first identity to the end.
await browser.contextualIdentities.move(otherIds, 0);
```

Another example, move "Personal" identity before "Work". Important note: this example assumes containers with these names to exist. This is often not the case in customized or localized (non-English) Firefox instances.

```js
let identities = await browser.contextualIdentities.query({});

// Find cookieStoreId of container with name "Personal".
let personalId = identities.find(ci => ci.name === "Personal")?.cookieStoreId;
if (!personalId) {
  throw new Error("Personal container not found");
}

// Find index of container with name "Work".
let workIndex = identities.findIndex(identity => identity.name === "Work");
if (workIndex === -1) {
  throw new Error("Work container not found!");
}
await browser.contextualIdentities.move(personalId, workIndex);
```

**Review by Rob--W — CHANGES_REQUESTED** [review]

File: files/en-us/mozilla/add-ons/webextensions/api/contextualidentities/move/index.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  position                         // integer
+)
+```
+
+### Parameters
+
+- `cookieStoreIds`
+  - : `string` or `array` of `string`. An ordered list of the contextual identity cookie store IDs to move.
+- `position`
+  - : `integer`. The position to move `cookieStoreIds` to in the list of contextual identities. `-1` indicates that the items are moved to the end of the list.
```
[the remark this answers is not in the document]
Comment (rebloor) [reply files/en-us/mozilla/add-ons/webextensions/api/contextualidentities/move/index.md:28]:
```suggestion
  - : `integer`. The position to move `cookieStoreIds` to in the list of contextual identities. Zero-based; `0` indicates the first position. `-1` indicates that the items are moved to the end of the list.
```

File: files/en-us/mozilla/add-ons/webextensions/api/contextualidentities/move/index.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+### Return value
+
+A [`Promise`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) that is fulfilled with an array of cookie store IDs representing the re-ordered list. If the request is for an invalid move or the contextual identities feature is not enabled, the promise is rejected.
```
[the remark this answers is not in the document]
Comment (rebloor) [reply files/en-us/mozilla/add-ons/webextensions/api/contextualidentities/move/index.md:32]:
```suggestion
A [`Promise`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) that is fulfilled when the contextual identities are reordered. If the request is for an invalid move or the contextual identities feature is not enabled, the promise is rejected.
```

File: files/en-us/mozilla/add-ons/webextensions/api/contextualidentities/move/index.md (hunk 1 above)
Comment (rebloor) [reply files/en-us/mozilla/add-ons/webextensions/api/contextualidentities/move/index.md]: Thanks for the suggestions @Rob--W

File: files/en-us/mozilla/add-ons/webextensions/api/contextualidentities/move/index.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+let personalId = identities.find((ci) => ci.name === "Personal")?.cookieStoreId;
+if (!personalId) {
+  throw new Error("Personal container not found");
+}
+
+// Find the index of the container with the name "Work".
+let workIndex = identities.findIndex((identity) => identity.name === "Work");
+if (workIndex === -1) {
+  throw new Error("Work container not found!");
+}
```
Comment (Rob--W) [inline files/en-us/mozilla/add-ons/webextensions/api/contextualidentities/move/index.md:77]: Edit to account for shifting by one:

```suggestion
// Find the index and ID of the container with the name "Personal".
let personalIndex = identities.findIndex((ci) => ci.name === "Personal");
if (personalIndex === -1) {
  throw new Error("Personal container not found");
}
let personalId = identities[personalIndex].cookieStoreId;

// Find the index of the container with the name "Work".
let workIndex = identities.findIndex((identity) => identity.name === "Work");
if (workIndex === -1) {
  throw new Error("Work container not found!");
}

if (personalIndex < workIndex) {
  // When the Personal identity moves, all following
  // identities shift to the left by one. To place
  // the Personal identity before the Work identity,
  // we should therefore subtract one.
  workIndex--;
}
```

**Review by Rob--W — APPROVED** [review]
Please apply the suggested edits before merging.

**Pull request merged**